scene/2d/tile_map.cpp

```diff
@@ -2759,114 +2906,114 @@ int TileMap::get_selected_layer() const {
-				// Physics.
-				bool in_editor = false;
+		case TileMap::NOTIFICATION_INTERNAL_PHYSICS_PROCESS: {
+			// Physics.
+			bool in_editor = false;
 #ifdef TOOLS_ENABLED
-				in_editor = Engine::get_singleton()->is_editor_hint();
+			in_editor = Engine::get_singleton()->is_editor_hint();
 #endif
-				if (is_inside_tree() && collision_animatable && !in_editor) {
-					// Update transform on the physics tick when in animatable mode.
-					last_valid_transform = new_transform;
-					set_notify_local_transform(false);
-					set_global_transform(new_transform);
-					set_notify_local_transform(true);
-				}
-			} break;
+			if (is_inside_tree() && collision_animatable && !in_editor) {
+				// Update transform on the physics tick when in animatable mode.
+				last_valid_transform = new_transform;
+				set_notify_local_transform(false);
+				set_global_transform(new_transform);
+				set_notify_local_transform(true);
+			}
+		} break;
 
-			case TileMap::NOTIFICATION_LOCAL_TRANSFORM_CHANGED: {
-				for (Ref<TileMapLayer> &layer : layers) {
-					layer->notify_local_xform_changed();
-				}
+		case NOTIFICATION_TRANSFORM_CHANGED: {
+			// Physics.
+			for (Ref<TileMapLayer> &layer : layers) {
+				layer->notify_tile_map_change(TileMapLayer::DIRTY_FLAGS_TILE_MAP_XFORM);
+			}
+		} break;
+
+		case TileMap::NOTIFICATION_LOCAL_TRANSFORM_CHANGED: {
+			for (Ref<TileMapLayer> &layer : layers) {
+				layer->notify_tile_map_change(TileMapLayer::DIRTY_FLAGS_TILE_MAP_LOCAL_XFORM);
+			}
 
-				// Physics.
-				bool in_editor = false;
+			// Physics.
+			bool in_editor = false;
 #ifdef TOOLS_ENABLED
-				in_editor = Engine::get_singleton()->is_editor_hint();
+			in_editor = Engine::get_singleton()->is_editor_hint();
 #endif
 
-				// Only active when animatable. Send the new transform to the physics...
-				if (is_inside_tree() && !in_editor && collision_animatable) {
-					// ... but then revert changes.
-					set_notify_local_transform(false);
-					set_global_transform(last_valid_transform);
-					set_notify_local_transform(true);
-				}
-			} break;
-		}
+			// Only active when animatable. Send the new transform to the physics...
+			if (is_inside_tree() && collision_animatable && !in_editor) {
+				// Store last valid transform.
+				new_transform = get_global_transform();
+
+				// ... but then revert changes.
+				set_notify_local_transform(false);
+				set_global_transform(last_valid_transform);
+				set_notify_local_transform(true);
+			}
+		} break;
 	}
 }
 
-void TileMap::queue_update_dirty_quadrants() {
-	if (pending_update || !is_inside_tree()) {
+#ifndef DISABLE_DEPRECATED
+// Deprecated methods.
+void TileMap::force_update(int p_layer) {
+	notify_runtime_tile_data_update(p_layer);
+	update_now();
+}
+#endif
+
+void TileMap::queue_internal_update() {
+	if (pending_update) {
 		return;
 	}
 	pending_update = true;
-	call_deferred(SNAME("_update_dirty_quadrants"));
+	call_deferred(SNAME("_internal_update"));
```

Comment:
```suggestion
	callable_mp(this, &TileMap::_internal_update).call_deferred();
```

And you don't need to bind it anymore.